src/cascadia/WindowsTerminal/TrayIcon.cpp

```diff
@@ -23,6 +23,37 @@ TrayIcon::~TrayIcon()
     RemoveIconFromTray();
 }
 
+void TrayIcon::CreateWindowProcess()
+{
+    WNDCLASSW wc{};
+    wc.hCursor = LoadCursor(nullptr, IDC_ARROW);
+    wc.hInstance = wil::GetModuleInstanceHandle();
+    wc.lpszClassName = L"TRAY_ICON_HOSTING_WINDOW_CLASS";
+    wc.style = CS_HREDRAW | CS_VREDRAW;
+    wc.lpfnWndProc = &TrayIcon::_WindowProc;
+    wc.hIcon = static_cast<HICON>(GetActiveAppIconHandle(true));
+    RegisterClass(&wc);
+
+    _trayIconHwnd = wil::unique_hwnd(CreateWindowW(wc.lpszClassName,
+                                                   wc.lpszClassName,
+                                                   WS_DISABLED,
+                                                   CW_USEDEFAULT,
+                                                   CW_USEDEFAULT,
+                                                   CW_USEDEFAULT,
+                                                   CW_USEDEFAULT,
+                                                   HWND_MESSAGE,
+                                                   nullptr,
+                                                   wc.hInstance,
+                                                   nullptr));
+
+    WINRT_VERIFY(_trayIconHwnd);
+}
+
+LRESULT CALLBACK TrayIcon::_WindowProc(HWND window, UINT message, WPARAM wparam, LPARAM lparam) noexcept
+{
+    return DefWindowProc(window, message, wparam, lparam);

```

Comment:
I think you can actually just literally set, `lpfnWndProc = DefWindowProcW` above! No need for all this ^^